### PR TITLE
Refactor update

### PR DESCRIPTION
**Pull request by iamnwi**
- refactor update promotion API
- remove unnecessary `print` in unit tests

### CODE REVIEW

File: service/service.py [hunk 1]
```diff
@@ -148,10 +148,33 @@ def get(self, promotion_id):
...
+
+        This endpoint will update a Promotion based the body that is posted
+        """
+        app.logger.info(
+            'Request to update promotion with promotion id {}'.format(promotion_id))
+        check_content_type('application/json')
+        promotion = Promotion.find(promotion_id)
+        if not promotion:
+            raise NotFound(
+                "Promotion with id '{}' was not found.".format(promotion_id))
```
Comment (reneesun) [inline service/service.py]: Just out of curiosity, why not use api.abort here?

Comment (iamnwi) [reply service/service.py]: I just reused the old code, not realizing that we have changed to use `api.abort`. Although the outcome of these two may be the same, I will change this to `api.abort` for consistency.

**Comment by iamnwi** [comment]
> LGTM. Happy thanksgiving!

Happy Thanksgiving!

**Comment by reneesun** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/a903ceab-af58-4d40-9de5-39501ad15e1a?env_id=ibm%3Ayp%3Aus-south "nyu-devops-fall19-promotion-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/5860bddd-fc68-4c80-afbc-55ea22efe279 "nyu-devops-fall19-promotion-pipeline") deployed [promotions](https://github.com/devops-fall-2019-promotions-squad/promotions/releases/tag/deploy-ibm-yp-us-south-rs6565-nyu.edu-dev-20191209-003910 "deploy-ibm-yp-us-south-rs6565-nyu.edu-dev-20191209-003910") to [dev](https://console.bluemix.net/apps/efcd7d31-ed8b-433d-b811-007435b814df?env_id=ibm:yp:us-south "ibm:yp:us-south:rs6565@nyu.edu:dev"), including commit 84f5a90c1b24462769f8eb16b19091656c53a743

**Comment by reneesun** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/a903ceab-af58-4d40-9de5-39501ad15e1a?env_id=ibm%3Ayp%3Aus-south "nyu-devops-fall19-promotion-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/5860bddd-fc68-4c80-afbc-55ea22efe279 "nyu-devops-fall19-promotion-pipeline") deployed [promotions](https://github.com/devops-fall-2019-promotions-squad/promotions/releases/tag/deploy-ibm-yp-us-south-rs6565-nyu.edu-prod-20191209-004350 "deploy-ibm-yp-us-south-rs6565-nyu.edu-prod-20191209-004350") to [prod](https://console.bluemix.net/apps/ce7c172b-4a93-4d4a-be2f-bbf7bd1fb4b7?env_id=ibm:yp:us-south "ibm:yp:us-south:rs6565@nyu.edu:prod"), including commit 84f5a90c1b24462769f8eb16b19091656c53a743

**Pull request merged**